File: Java/src/1603_DesignParkingSystem/DesignParkingSystem.java
class ParkingSystem {
    private int big ;
    private int medium;
    private int small;
    public ParkingSystem(int big, int medium, int small) {
        this.big = big;
        this.medium = medium;
        this.small = small;
    }
    
    public boolean addCar(int carType) {
        if(carType == 1){
            if(this.big-1 < 0) return false;
            else {
                this.big--;
                return true;
            }
        }
        else if(carType == 2){
            if(this.medium-1 < 0) return false;
            else {
                this.medium--;
                return true;
            }
        }
        else{
            if(this.small-1 < 0) return false;
            else {
                this.small--;
                return true;
            }
            
        }
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem obj = new ParkingSystem(big, medium, small);
 * boolean param_1 = obj.addCar(carType);
 */

File: golang/1603_DesignParkingSystem/designParkingSystem.go
package designparkingsystem

type ParkingSystem struct {
	big    int
	medium int
	small  int
}

func Constructor(big int, medium int, small int) ParkingSystem {
	return ParkingSystem{
		big: big, medium: medium, small: small,
	}
}

func (this *ParkingSystem) AddCar(carType int) bool {
	if carType == 1 {
		if this.big > 0 {
			this.big--
			return true
		}
	} else if carType == 2 {
		if this.medium > 0 {
			this.medium--
			return true
		}
	} else if carType == 3 {
		if this.small > 0 {
			this.small--
			return true
		}
	}
	return false
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * obj := Constructor(big, medium, small);
 * param_1 := obj.AddCar(carType);
 */
